builtin/mainmenu/textures.lua

```diff
@@ -145,6 +145,27 @@ function mm_texture.set_game(identifier,gamedetails)
 	
 	local path = gamedetails.path .. DIR_DELIM .."menu" ..
 									 DIR_DELIM .. identifier .. ".png"
+
+	if core.setting_getbool("random_menu_images") == true then
+		local count = 1
+		for i = 1, 9 do
+			local check_path = path .. "." .. i
+			local file = io.open(check_path,"r")
+			if file == nil then
+				count = i - 1
+				break
+			else
+				io.close(file)
+			end
+		end
+		local new_path = path .. "." .. math.random(1, count)
+		local exists = io.open(new_path,"r")
+		if exists ~= nil and math.random(1, (count+1)) ~= 1 then
```

Comment:
Because without that, either background.png or background.png.1 will not be shown (depending on other changes made), which, imho, is not ideal behaviour.